src/test_workflow/integ_test/integ_test_suite.py

```diff
@@ -114,7 +116,8 @@ def custom_node_endpoint_encoder(node_endpoint: NodeEndpoint) -> dict:
 
     def execute_integtest_sh(self, endpoint: str, port: int, security: bool, test_config: str) -> int:
         cluster_endpoint_port = [ClusterEndpoint("cluster1", [NodeEndpoint(endpoint, port, 9300)], [])]
-        return self.multi_execute_integtest_sh(cluster_endpoint_port, security, test_config)
+        script = ScriptFinder.find_integ_test_script("functionalTestDashboards", "https://github.com/opensearch-project/opensearch-dashboards-functional-test.git")
```

Comment:
This will break for opensearch plugins. functional test only applies to dashboards. Also the I don't see `ref` anywhere, this will by default clone main. You need to get the repo link and ref from input manifest, https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.6.0/opensearch-dashboards-2.6.0.yml#L13-L15

Reply:
I see the function execute_integtest_sh in used only for OSD which calls multi_execute_integtest_sh, whereas OS directly uses multi_execute_integtest_sh.

Reply:
Yeah we need to change that behavior. Can you fix that as well? Adding @peterzhuamazon @prudhvigodithi for more input

Reply:
Use the git lib in build repo to checkout ftrepo, do not need to use a find script call like this, because if cwd already have ftrepo content, you can just call the find script for integtest.sh and it will find it in root.

Reply:
Sure Peter, started working on this change. Trying to fetch the integtest.sh from ftrepo based on ref branch from input manifest.